src/Vault.sol

```diff
@@ -873,7 +886,7 @@ contract Vault is EIP1967Admin, VaultAccessControl, IERC721Receiver, ICDP, Multi
         address target
     ) internal {
         if (IERC20(targetToken).allowance(address(this), target) < amount) {
```

Comment:
Seems that this "if" is no longer needed, since we are setting an exact approval each time. We can also get rid of one-line _checkAllowance function then.